lib/Value/Real.pm

```diff
@@ -154,6 +154,29 @@ sub compare {
       }
       return 0;
     }
+    if ($self->getFlag('tolType') eq 'digits-2') {
+      return 0 if (abs($a) < $zeroLevel and abs($b) < $zeroLevel);
+      return $a <=> $b if (abs($a) < $zeroLevel or abs($b) < $zeroLevel);
+      $tolerance = -log($tolerance)/log(10) if ($tolerance > 0 and $tolerance < 1);
+      my $digits = (1 > int($tolerance) ? 1 : int($tolerance + 0.5)) - 1;
```

Comment:
Is the subtraction of 1 at the end here intentional? I noticed it because it's a change from the previous version. If `$tolerance` has somehow ended up being 1.00001, then this makes `$digits` be 0.

My plan is to merge this into my branch, delete the original, and rename this one `digits`. Also, I think it will be cleaner if I change `tolDigits` to mean the _additional_ number of digits to check. So instead of "6" it would be "3" here. So if you agree that the subtraction by 1 should not happen, don't make an edit. I would do that when I am following up.

Reply:
Yes, this is intentional.  Here, `$digits` is the number of decimal digits in the exponential form, so there is always one digit before the decimal point.  Wo we need to compensate for that.

Reply:
OK, but does that mean that `(1 > int($tolerance) ? 1` would be better as `(1 > int($tolerance) ? 0`?

Reply:
Since 1 is subtracted from the end result of the ` ? : ` operation, no.  But if you want to change it to 

```
(1 > int($tolerance) ? 0 : int($tolerance + 0.5) - 1);
```

you could.

Reply:
Doh. I did not parse the subtraction as happening after the trinary operator. Thanks for making that clear.